youtube_dl/extractor/motherless.py

```diff
@@ -85,18 +85,25 @@ def _real_extract(self, url):
             or 'http://cdn4.videos.motherlessmedia.com/videos/%s.mp4?fs=opencloud' % video_id)
         age_limit = self._rta_search(webpage)
         view_count = str_to_int(self._html_search_regex(
-            (r'>(\d+)\s+Views<', r'<strong>Views</strong>\s+([^<]+)<'),
+            (r'>([\d,.]+)\s+Views<',  # 1,234,567 Views
+             r'<strong>Views</strong>\s+([^<]+)<'),
             webpage, 'view count', fatal=False))
         like_count = str_to_int(self._html_search_regex(
-            (r'>(\d+)\s+Favorites<', r'<strong>Favorited</strong>\s+([^<]+)<'),
+            (r'>([\d,.]+)\s+Favorites<',  # 1,234 Favorites
+             r'<strong>Favorited</strong>\s+([^<]+)<'),
             webpage, 'like count', fatal=False))
 
         upload_date = self._html_search_regex(
             (r'class=["\']count[^>]+>(\d+\s+[a-zA-Z]{3}\s+\d{4})<',
+             r'class=["\']count[^>]+>(\d+d\s+[aA]go)<',  # 1d ago
+             r'class=["\']count[^>]+>(\d+h\s+[aA]go)<',  # 20h ago
              r'<strong>Uploaded</strong>\s+([^<]+)<'), webpage, 'upload date')
-        if 'Ago' in upload_date:
+        if re.search(r'd\s+[aA]go', upload_date):
             days = int(re.search(r'([0-9]+)', upload_date).group(1))
             upload_date = (datetime.datetime.now() - datetime.timedelta(days=days)).strftime('%Y%m%d')
+        elif re.search(r'h\s+[aA]go', upload_date):
+            hours = int(re.search(r'([0-9]+)', upload_date).group(1))
+            upload_date = (datetime.datetime.now() - datetime.timedelta(hours=hours)).strftime('%Y%m%d')
```

Comment:
DRY.

Reply:
I fixed this one too, the code is a bit longer though not sure it is clearer.

Reply:
I was not sure about splitting in a sub-function. If you have a format you prefer, just point me to an example in the repo and I will update it again.